### PR TITLE
[fast-ut] [reduced-it] Add collection expression corner-case coverage [databricks]

### LINKED ISSUE #15784: [TEST] Cover ArrayPosition and MapZipWith Binary/Decimal corner cases
## Coverage gap

The Python integration tests do not currently cover these supported
expression/type combinations:

- `array_position` with `BinaryType` array elements
- `map_zip_with` with nested Decimal64 map keys/values
- `map_zip_with` with nested Decimal128 map keys/values

These cells are part of the audited expression/type corner-case manifest but
are absent from the existing literal `data_gen` parametrizations.

## Intended test scope

- Extend the existing `test_array_position` parametrization with `BinaryGen()`.
- Extend the existing `test_map_zip_with` parametrization with literal
  `MapGen(DecimalGen(...), DecimalGen(...))` entries for precision 12 and 20.
- Require a GPU project in the new parameters.
- Mark the two Decimal `map_zip_with` parameters `xfail` against the confirmed
  correctness issue [#15783](https://github.com/NVIDIA/cudf-spark/issues/15783) until the production bug is fixed.

This is coverage-only work: no production behavior or expected result is being
changed. An `xfail` counts as definition coverage while keeping the known
runtime correctness debt explicit.

### PR DESCRIPTION
**Pull request by wjxiz1992**
**JaCoCo production line coverage: not fully measurable locally — `sql-plugin` is N/A because the shim-350/b23 baseline has 3 class-ID mismatches; measured compatible scope: `sql-plugin subset +0`, `delta-stub +0`, `iceberg +0`, `shuffle-plugin +0`, `udf-compiler +0`** (anchor `01ad3b33ff93c3a069aec12caad5dedac39eca7c`, Scala 2.12, shim 350, b23)

Fixes #15784.
Refs #15783.

### Description

This adds missing integration-test definition coverage for collection expression/type combinations:

- extend the existing `test_array_position` matrix with a literal `BinaryGen()` case;
- extend the existing `test_map_zip_with` matrix with non-null DECIMAL(12,2) and DECIMAL(20,2) map key/value cases;
- require `GpuProjectExec` for the added parameters.

The two DECIMAL cases reproduce the correctness defect tracked by #15783 and are marked `xfail` so the coverage definition lands without hiding the unresolved Milestone 2 work. There is no user-facing behavior change.

AI assistance: The change and PR description were prepared with Codex assistance and reviewed by the author before submission.

Validation:

- Spark 3.3 dist build: `BUILD SUCCESS`
- Spark 3.3 integration-test harness build: `BUILD SUCCESS`
- Spark 3.3 / Python 3.10 focused integration test: `1 passed, 2 xfailed, 1437 deselected`
- Spark 3.5.0 / shim-350 b23 focused integration test: `1 passed, 2 xfailed, 1436 deselected`

The full shim-350 `sql-plugin` delta is not reported because its nightly baseline is bytecode-incompatible for three classes, and the PR exec reaches one of those classes. The compatible 1850-class subset and the other published report groups each measured `+0` production lines. The shim-350 report contains only the Delta stub group, so it is not evidence of real Delta implementation coverage.

### Checklists

Documentation
- [ ] Updated for new or modified user-facing features or behaviors
- [x] No user-facing change

Testing
- [x] Added or modified tests to cover new code paths
- [ ] Covered by existing tests
      (Please provide the names of the existing tests in the PR description.)
- [ ] Not required

Performance
- [ ] Tests ran and results are added in the PR description
- [ ] Issue filed with a link in the PR description
- [x] Not required

### CODE REVIEW

**Comment by wjxiz1992** [comment]
build

File: integration_tests/src/main/python/map_test.py [hunk 1]
```diff
@@ -883,7 +911,22 @@ def do_it(spark):
     # ANSI mode is disabled since this test verifies the behaviour of map_zip_with and the evaluation of the associated lambda. 
     # Exceptions during overflow conditions are tested in the arithmetic-ops tests.
     # Not using @disable_ansi_mode because of https://github.com/NVIDIA/spark-rapids/issues/13214.  Using explicit setting instead.
-    assert_gpu_and_cpu_are_equal_collect(do_it, conf={'spark.sql.ansi.enabled': False})
+    try:
+        assert_gpu_and_cpu_are_equal_collect(do_it, conf={'spark.sql.ansi.enabled': False})
+    except AssertionError as error:
```
Comment (thirtiseven) [inline integration_tests/src/main/python/map_test.py:940]: This catches every result-comparison failure for Decimal inputs and translates it into the expected failure for #15783. `assert_gpu_and_cpu_are_equal_collect` raises `AssertionError` for any output mismatch, including row shape and unrelated differences in `n`, `one`, or `indexed`, so a new regression could remain hidden as an XFAIL. Could we restrict this translation to the known null mismatch in `ident1`/`ident2`, or isolate those expressions in a dedicated xfailed test?

**Comment by wjxiz1992** [comment]
build

File: integration_tests/src/main/python/map_test.py (hunk 1 above)
Comment (wjxiz1992) [reply integration_tests/src/main/python/map_test.py:940]: Updated.

**Review by thirtiseven — APPROVED** [review]
